docs/code/IDataViewTypeSystem.md

```diff
@@ -0,0 +1,844 @@
+values. For example, the standard conversion from `TX` to `R8` maps the empty
+text value to the value zero. Note that the empty text value is distinct from
+the missing text value, as discussed next.
+
+### Missing Value
+
+Most of the standard primitive types support the notion of a missing value. In
+particular, the text type, floating-point types, signed integer types, and key
+types all have an internal representation of missing. We follow R's lead and
+denote such values as `NA`.
+
+Unlike R, the standard primitive types do not distinguish between missing and
+invalid. For example, in floating-point arithmetic, computing zero divided by
+zero, or infinity minus infinity, produces an invalid value known as a `NaN`
+(for Not-a-Number). R uses a specific `NaN` value to represent its `NA` value,
+with all other `NaN` values indicating invalid. The IDataView standard
+floating-point types do not distinguish between the various `NaN` values,
+treating them all as missing/invalid.
+
+A standard conversion from a source type with `NA` to a destination type with
+`NA` maps `NA` to `NA`. A standard conversion from a source type with `NA` to
+a destination type without `NA` maps `NA` to the default value of the
+destination type. For example, converting a text `NA` value to `R4` produces a
+`NaN`, but converting a text `NA` to `U4` results in zero. Note that this
+specification does not address diagnostic user messages, so, in certain
+environments, the latter situation may generate a warning to the user.
+
+Note that a vector type does not support a representation of missing, but may
+contain `NA` values of its item type. Generally, there is no standard
+mechanism faster than O(N) for determining whether a vector with N items
+contains any missing values.
+
+For further details on missing value representations, see the sections
+detailing the particular standard primitive types.
+
+### Vector Representations
+
+Values of a vector type may be represented either sparsely or densely. A
+vector type does not mandate denseness or sparsity, nor does it imply that one
+is favored over the other. A sparse representation is semantically equivalent
+to a dense representation having the suppressed entries filled in with the
+*default* value of the item type. Note that the values of the suppressed
+entries are emphatically *not* the missing/`NA` value of the item type, unless
+the missing and default values are identical, as they are for key types.
+
+### Metadata
+
+A column in an `ISchema` can have additional column-wide information, known as
+metadata. For each string value, known as a metadata kind, a column may have a
+value associated with that metadata kind. The value also has an associated
+type, which is a compatible column type.
+
+For example:
+
+* A column may have a `BL` valued piece of metadata associated with the string
+`IsNormalized` indicating whether the column can be interpreted as a label.
```

Comment:
I find this example confusing. Are we saying a user named the column `IsNormalized`, or is the name of the metadata `IsNormalized`. Why would either of these tell if the column can be a label? (perhaps regression for linear models?)

Perhaps: "* A column may have a `BL` valued piece of metadata (along with its associated string `IsNormalized` to name it) indicating the column has already been normalized, and won't require additional normalization before certain learners or transforms."

Reply:
This entire sentence is absolute nonsense so I'm just going to rewrite it. Normalization has nothing to do with labels. Also I'm not sure why we're saying "Metadata is associated with the string..." Metadata just has names, period.

Hello, I am associated with the string Tom Finley. :D :D

---
In reply to: [189046936](https://github.com/dotnet/machinelearning/pull/173#discussion_r189046936) [](ancestors = 189046936)

Reply:
Thank you great person associated with the string, "Tom Finley"!